.github/workflows/CD.yml

```diff
@@ -315,14 +317,24 @@ jobs:
       - name: Restore cached packages
         uses: actions/cache@v2
         with:
-          key: "${{ env.PACKAGE_CACHE_KEY }}"
+          key: "${{ env.TESTING_FRAMEWORK_REPO }}"
           path: build/packages
 
       - name: Checkout the sha
         uses: actions/checkout@v2
         with:
           ref: ${{ github.event.inputs.sha }}
           path: adot
+
+      #Since the tools from workflow are always uptodate and to keep it up with the github workflow, we need to check out the main branch
+      #And use the tools from there instead of the commited sha branch to avoid commit additional code for the commited sha's branch
+      #Documentation on side by side checkout: https://github.com/actions/checkout#checkout-multiple-repos-side-by-side
+
+      - name: Checkout main branch to get workflow tools
+        uses: actions/checkout@v2
+        with:
+          ref:  "${{ env.MAIN_BRANCH }}"
```

Comment:
I don't think you need to make this an environment variable.

Reply:
Reasonable. Fixed it.